### ISSUE TITLE
a11y: Project discipline buttons use invalid ARIA state

### ISSUE THREAD

**Issue by eatyourgreens** [zooniverse/front-end-monorepo#7197]
https://github.com/zooniverse/front-end-monorepo/blob/d78ac1576bff1efe8b601787fc85d5cb7b44d9d8/packages/lib-content/src/screens/Projects/components/Projects/components/DisciplineSelect.jsx#L108-L113

Project discipline buttons use `aria-selected` here, but `aria-selected` isn't an allowed state for the `button` role. Discipline buttons should use a state that's supported on buttons, such as `aria-pressed` or `aria-current`, otherwise the selected discipline won't be announced in JAWS or NVDA.

Here are screenshots showing the bug in the Chrome accessibility tree:

Invalid `aria-selected=true` exposes no extra button state to screen readers.

<img width=800 alt="Chrome dev tools showing accessibility tree properties for a button with aria-selected." src="https://github.com/user-attachments/assets/1bcacff0-5c04-4852-8a70-0130537a2559" />

Valid `aria-pressed=true` exposes `pressed: true` to screen readers.

<img width="800" alt="Chrome dev tools showing accessibility tree properties for a button with aria-pressed." src="https://github.com/user-attachments/assets/29a1a6bd-eb5b-4bf0-b613-a7201ddef8fe" />

## References
- [`aria-selected` state](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-selected).
- [`button` role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/button_role).

**Comment by eatyourgreens** [comment]
Updated to add screenshots of the bug in the Chrome accessibility tree.

This is a WCAG level A failure, since a blind user can't tell which project category is selected, if any.
